### PR TITLE
Feature display single animal

### PR DESCRIPTION
**Pull request by irenepapaspyratos**

### CODE REVIEW

File: src/components/Animal/Animal.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		<article>
+			<header>
+				<h2>{animal.name}</h2>
+			</header>
+
+			<section>
+				<h3>Description</h3>
+				<p>{animal.description}</p>
+			</section>
+
```
Comment (PReichetanz) [inline src/components/Animal/Animal.jsx:14]: I would delete empty lines inside a return statement (I think Prettier normally would).

File: src/components/Animal/Animal.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+					{spots
+						.filter(spot => {
+							return animal.spots.includes(spot.name) && spot.name;
+						})
+						.map(spot => {
+							return <li key={spot.id}>{spot.name}</li>;
+						})}
+				</ul>
+			</section>
+
```
Comment (PReichetanz) [inline src/components/Animal/Animal.jsx:27]: Same here.

File: src/components/Animal/Animal.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import PropTypes from 'prop-types';
+
+export default function Spot({ spots, animal, organizers }) {
+	return (
+		<article>
+			<header>
+				<h2>{animal.name}</h2>
+			</header>
+
+			<section>
```
Comment (PReichetanz) [inline src/components/Animal/Animal.jsx:10]: You could think about outsourcing your three sections to different components in order to make your Animal.jsx more readable. But you don't have to write a story for this kind of refactoring, but can create an issue on your project board where you can link the PR for these changes.

File: src/components/Animal/Animal.spec.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
+import Animal from './Animal';
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+
+describe('Single Animal-Display', () => {
+	it("renders every important detail of animal & the names of spots where animal lives & NOT the names of spots where animal is not living  & the names of organizers who arrange diving with animal & NOT the names of organiZers who don't arrange diving with animal ", () => {
```
Comment (PReichetanz) [inline src/components/Animal/Animal.spec.js:6]: I think it is really hard to use this test in real life, i.e. if it fails to figure out what's wrong.
I would separate it in different `it` statements (what is rendered and what is not); you can create variables above with your fake data which you use in line 9 for the `spots` prop (and the other props as well) and thus hand over this variable every time your Animal component is rendered in the different `it` statements.
By doing so, you don't have to write the fake data for every single `it`.

File: src/components/Animal/Animal.spec.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+					},
+				]}
+			/>
+		);
+
+		const nameSpot1 = screen.queryByText(/Bimini/i);
+		const nameSpot2 = screen.queryByText(/Gran Bahama/i);
+		const nameSpot3 = screen.queryByText(/Socorro Islands/i);
+
+		const animalDetail1 = screen.queryByText(/Dolphin/i);
```
Comment (PReichetanz) [inline src/components/Animal/Animal.spec.js:92]: I think the names are rendered as `h2`, right? You can thus use `getByRole` with a heading role in order to be more precise what you are looking for.

File: src/components/Spot/Spot.spec.js [hunk 6]
```diff
@@ -3,7 +3,7 @@ import '@testing-library/jest-dom';
 import { render, screen } from '@testing-library/react';
 
 describe('Single Spot-Display', () => {
-	it('renders every important detail of spot & the names of animals living there & NOT the names of animals not living there & the names of organiz', () => {
+	it('renders every important detail of spot & the names of animals living there & NOT the names of animals not living there & the names of organizers at this location & NOT the names of organizers not located at this location', () => {
```
Comment (PReichetanz) [inline src/components/Spot/Spot.spec.js:6]: Here I would separate the test as well.

File: src/components/ui/Animal/Animal.stories.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+const props = {
+	spots: [
+		{
+			id: '1',
+			name: 'Bimini',
+			country: 'Bahamas',
+			geoTag: { latitude: '25.703188', longitude: '-79.281459' },
+			description: 'nice and warm',
+		},
+		{
```
Comment (PReichetanz) [inline src/components/ui/Animal/Animal.stories.js:26]: Since you are using the same fake data here as well as in your Animal.spec.js, why not outsourcing this variable to another file and simply import it in bot spec files?

**Review by PReichetanz — COMMENTED** [review]
Nice work, I just have some minor issues to think about 🚀

File: src/components/Animal/Animal.jsx (hunk 1 above)
Comment (irenepapaspyratos) [reply src/components/Animal/Animal.jsx:14]: I know, but as long as we don't have a corresponding explicit requirement for our code, I prefer to keep my extra spacing to have a better overview.

File: src/components/Animal/Animal.jsx (hunk 2 above)
Comment (irenepapaspyratos) [reply src/components/Animal/Animal.jsx:27]: Same here.

File: src/components/Animal/Animal.jsx (hunk 3 above)
Comment (irenepapaspyratos) [reply src/components/Animal/Animal.jsx:10]: I will keep this in mind and think about it as soon as I am working with the 'real' data, because there will be less logic needed then.

File: src/components/ui/Animal/Animal.stories.js (hunk 7 above)
Comment (irenepapaspyratos) [reply src/components/ui/Animal/Animal.stories.js:26]: Unfortunately it's not the same data.

File: src/components/Spot/Spot.spec.js [hunk 8]
```diff
@@ -87,21 +87,179 @@ describe('Single Spot-Display', () => {
...
+		);
+
+		const nameAnimal1 = screen.queryByText(/Dolphin/i);
+		const nameAnimal2 = screen.queryByText(/Shark/i);
 
 		expect(nameAnimal1).toBeInTheDocument();
 		expect(nameAnimal2).not.toBeInTheDocument();
+	});
+
+	it('renders the names of organizers at this location & NOT the names of organizers not located at this location', () => {
```
Comment (PReichetanz) [inline src/components/Spot/Spot.spec.js:182]: Could you rephrase like
```suggestion
	it('renders only the names of organizers at this location', () => {
```
?

File: src/components/Spot/Spot.spec.js [hunk 9]
```diff
@@ -87,21 +87,179 @@ describe('Single Spot-Display', () => {
...
-		const nameOrganizer3 = screen.queryByText(/Pacific Fleet/i);
-
 		expect(spotDetail1).toBeInTheDocument();
 		expect(spotDetail2).toBeInTheDocument();
 		expect(spotDetail3).toBeInTheDocument();
 		expect(spotDetail4).toBeInTheDocument();
 		expect(spotDetail5).toBeInTheDocument();
+	});
+
+	it('renders the names of animals living there & NOT the names of animals not living there', () => {
```
Comment (PReichetanz) [inline src/components/Spot/Spot.spec.js:97]:
```suggestion
	it('renders only the names of animals living there', () => {
```

File: src/components/Animal/Animal.spec.js [hunk 10]
```diff
@@ -85,29 +85,195 @@ describe('Single Animal-Display', () => {
...
+		const nameSpot1 = screen.queryByText(/Bimini/i);
+		const nameSpot2 = screen.queryByText(/Gran Bahama/i);
+		const nameSpot3 = screen.queryByText(/Socorro Islands/i);
+
+		expect(nameSpot1).toBeInTheDocument();
+		expect(nameSpot2).toBeInTheDocument();
+		expect(nameSpot3).not.toBeInTheDocument();
+	});
+
+	it("renders the names of organizers who arrange diving with animal & NOT the names of organizers who don't arrange diving with animal ", () => {
```
Comment (PReichetanz) [inline src/components/Animal/Animal.spec.js:192]:
```suggestion
	it("renders only the names of organizers who arrange diving with animal", () => {
```

File: src/components/Animal/Animal.spec.js [hunk 11]
```diff
@@ -85,29 +85,195 @@ describe('Single Animal-Display', () => {
...
-		expect(nameSpot3).not.toBeInTheDocument();
-
 		expect(animalDetail1).toBeInTheDocument();
 		expect(animalDetail2).toBeInTheDocument();
 		expect(animalDetail3).toBeInTheDocument();
 		expect(animalDetail4).toBeInTheDocument();
 		expect(animalDetail5).toBeInTheDocument();
+	});
+
+	it('renders the names of spots where animal lives & NOT the names of spots where animal is not living ', () => {
```
Comment (PReichetanz) [inline src/components/Animal/Animal.spec.js:101]:
```suggestion
	it('renders only the names of spots where animal lives', () => {
```

**Review by PReichetanz — COMMENTED** [review]
Looks good, just had one idea about the `it` statement because I don't think you have to add the "NOT" part if you add an "only" in the first place ;)

**Review by PReichetanz — APPROVED** [review]

**Comment by UrsulaMontag** [comment]
Hey Irene,
it looks very good to me.
